packages/EDA/wasm/EDAAPI.js

```diff
@@ -129,3 +129,48 @@ export async function _trainAndAnalyzeLSSVMInWebWorker(gamma, kernel, kernelPara
   });
 }
 
+export function _fitLinearRegressionParamsWithDataNormalizing(features, featureAvgs, featureStdDevs, targets, targetsAvg, targetsStdDev, paramsCount) {
+  return callWasm(EDA, 'fitLinearRegressionParamsWithDataNormalizing', [features, featureAvgs, featureStdDevs, targets, targetsAvg, targetsStdDev, paramsCount]);
+}
+
+export async function _fitLinearRegressionParamsWithDataNormalizingInWebWorker(features, featureAvgs, featureStdDevs, targets, targetsAvg, targetsStdDev, paramsCount) {
+  return new Promise((resolve, reject) => {
+    const worker = new Worker(new URL('../wasm/workers/fitLinearRegressionParamsWithDataNormalizingWorker.js', import.meta.url));
+    worker.postMessage(getCppInput(EDA['fitLinearRegressionParamsWithDataNormalizing'].arguments,[features, featureAvgs, featureStdDevs, targets, targetsAvg, targetsStdDev, paramsCount]));
+    worker.onmessage = function(e) {
+      worker.terminate();
+      resolve(getResult(EDA['fitLinearRegressionParamsWithDataNormalizing'], e.data));
+    }
+  });
+}
+
+export function _fitLinearRegressionParams(features, targets, paramsCount) {
+  return callWasm(EDA, 'fitLinearRegressionParams', [features, targets, paramsCount]);
+}
+
+export async function _fitLinearRegressionParamsInWebWorker(features, targets, paramsCount) {
```

Comment:
Oh nice, didn't even know we were doing it in the web worker! But linear regressions is very fast - it's up to you, but it's fine to do it in the main thread I think. I guess do it the way it makes sense to keep the code simple.

Reply:
This file is generated automatically by our c++-to-wasm export script. For each exported c++-function, this script generates two versions: in web worker call and in the main thread. After this, we can choose the preferable approach.
Currently, in linear regression, we use the second one